src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -71,31 +71,31 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetProductName(char * buf
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetFirmwareRevisionString(char * buf, size_t bufSize)
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSoftwareVersionString(char * buf, size_t bufSize)
 {
-    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_FIRMWARE_REVISION_STRING), CHIP_ERROR_BUFFER_TOO_SMALL);
-    strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_FIRMWARE_REVISION_STRING);
+    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_REVISION_STRING), CHIP_ERROR_BUFFER_TOO_SMALL);
+    strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_REVISION_STRING);
     return CHIP_NO_ERROR;
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize, size_t & serialNumLen)
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize)
 {
     CHIP_ERROR err;
-    err = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
+    size_t serialNumLen = 0;
+    err                 = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
+    ReturnErrorCodeIf(serialNumLen > bufSize, err = CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(err == CHIP_NO_ERROR && buf[serialNumLen] != 0, err = CHIP_ERROR_INVALID_STRING_LENGTH);
```

Comment:
This breaks the `CHIP_DEVICE_CONFIG_TEST_SERIAL_NUMBER` codepath, where err is in fact supposed to be CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND, no?